pkg/api/handlers/libpod/containers.go

```diff
@@ -443,12 +443,47 @@ func UpdateContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	options := &handlers.UpdateEntities{Resources: &specs.LinuxResources{}}
-	if err := json.NewDecoder(r.Body).Decode(&options.Resources); err != nil {
+	options := &handlers.UpdateEntities{}
+	if err := json.NewDecoder(r.Body).Decode(&options); err != nil {
 		utils.Error(w, http.StatusInternalServerError, fmt.Errorf("decode(): %w", err))
 		return
 	}
-	err = ctr.Update(options.Resources, restartPolicy, restartRetries)
+
```

Comment:
can you just call (ic *ContainerEngine) ContainerUpdate() from like many other endpoints do to avoid duplication